blivet/fcoe.py

```diff
@@ -110,27 +111,69 @@ def addSan(self, nic, dcb=False, auto_vlan=True):
         log.info("Activating FCoE SAN attached to %s, dcb: %s autovlan: %s",
                  nic, dcb, auto_vlan)
 
-        util.run_program(["ip", "link", "set", nic, "up"])
-
         rc = 0
+        out = ""
+        timeout = 60
         error_msg = ""
         if dcb:
+
             self._startLldpad()
-            util.run_program(["dcbtool", "sc", nic, "dcb", "on"])
-            util.run_program(["dcbtool", "sc", nic, "app:fcoe",
-                                                "e:1", "a:1", "w:1"])
-            rc, out = util.run_program_and_capture_output(["fipvlan", "-c", "-s", "-f",
-                                               "-fcoe", nic])
+            while timeout > 0:
+                rc, out = util.run_program_and_capture_output(
+                    ["lldptool", "-p"])
+                if not rc:
+                    break
+                log.info("Waiting for lldpad to be ready")
```

Comment:
these are going to clutter up the logs, maybe only log if the timeout is exceeded. while / else could be used for that.